python/private/py_toolchain_suite.bzl

```diff
@@ -100,14 +100,19 @@ def py_toolchain_suite(*, prefix, user_repository_name, python_version, set_pyth
         **kwargs
     )
 
+    is_precompile_enabled = Label("//python/config_settings:is_precompile_toolchain_enabled")
     native.toolchain(
         name = "{prefix}_py_exec_tools_toolchain".format(prefix = prefix),
         toolchain = "@{user_repository_name}//:py_exec_tools_toolchain".format(
             user_repository_name = user_repository_name,
         ),
         toolchain_type = EXEC_TOOLS_TOOLCHAIN_TYPE,
-        # The target settings capture the Python version
-        target_settings = target_settings,
+        # The target settings capture the Python version, the select works like a
+        # (target_settings AND is_precompile_enabled)
+        target_settings = select({
+            is_precompile_enabled: target_settings,
+            "//conditions:default": [is_precompile_enabled],
```

Comment:
I can't figure out what the conditions:default branch here is doing.

It's saying...

If the --precompile_toolchain=enabled, then use target_settings for the toolchain.

Otherwise, use --precompile_toolchain=enabled for the target settings. i.e. this toolchain will only match if --precompile_toolchain=enabled, but that's covered by the condition above.

is this a copy/paste error?

Reply:
It is not, I whish I could write:
```
target_settings = select({
    is_precompile_enabled: target_settings,
    "//conditions:default": ["@platforms//:incompatible"],
})
```

However, that does not work, or maybe I am missing something. The code above is the simplest way to have a list of `target_settings` that is not empty and that will ensure that nothing will be matched.

Reply:
Huh. Yeah, that code is exactly what I was going to rewrite it to. It doesn't work? What error does it give?

Reply:
Idea: maybe because `@platforms//:incompatible` is a constraint-thing (from the `constraint()` rule, which are like...a different subtly of constraint that config settings :shrug:) instead of config_setting? I'll give that a try.

Reply:
Ah. I see. That's annoying. `target_settings` apparently can only accept config_settings, so the incompatible target can't be used. Meanwhile, `target_compatible_with` can't be used because its nonconfigurable.

So I get the trick the original code was doing. Clever!